src/persist/src/indexed/trace.rs

```diff
@@ -289,6 +289,16 @@ impl Trace {
         // TODO: actually clear the unwanted batches from the blob storage
         let size_bytes = blob.set_trace_batch(key.clone(), new_batch)?;
 
+        // Only upgrade the compaction level if we know this new batch represents
+        // an increase in data over both of its parents so that we we know we need
+        // even more additional batches to amortize the cost of compacting it in
+        // the future.
+        let merged_level = if size_bytes > first.size_bytes && size_bytes > second.size_bytes {
+            first.level + 1
+        } else {
+            first.level
```

Comment:
I think it's worth adding a test for this. Compaction logic tests will quickly get unwieldy, but this is still straightforward enough.

Eventually, def not now, we should brainstorm how to write a more generalized compaction test framework that tests it at a higher level (e.g. model the mz_metrics workload and assert on how many batches come in before stuff cancels out/assert an upper bound on the total size of the trace). Then we can model different workloads with that and test new compactions strategies vs all of them

Reply:
done + cleaned up some wonky antichain handling leftover from before!